crates/turborepo-env/src/platform.rs

```diff
@@ -0,0 +1,104 @@
+}
+
+impl Default for PlatformEnv {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+const TURBO_PLATFORM_ENV_KEY: &str = "TURBO_PLATFORM_ENV";
+const TURBO_PLATFORM_ENV_DISABLED_KEY: &str = "TURBO_PLATFORM_ENV_DISABLED";
+
+impl PlatformEnv {
+    pub fn new() -> Self {
+        let env_keys = std::env::var(TURBO_PLATFORM_ENV_KEY)
+            .unwrap_or_default()
+            .split(',')
+            .filter(|s| !s.is_empty())
+            .map(|s| s.to_string())
+            .collect();
+
+        Self { env_keys }
+    }
+
+    pub fn disabled() -> bool {
+        let turbo_platform_env_disabled =
+            std::env::var(TURBO_PLATFORM_ENV_DISABLED_KEY).unwrap_or_default();
+        turbo_platform_env_disabled == "1" || turbo_platform_env_disabled == "true"
+    }
+
+    pub fn validate(&self, execution_env: &EnvironmentVariableMap) -> Vec<String> {
+        if Self::disabled() {
+            return vec![];
+        }
+
+        self.diff(execution_env)
+    }
+
+    pub fn diff(&self, execution_env: &EnvironmentVariableMap) -> Vec<String> {
+        self.env_keys
+            .iter()
+            .filter(|key| !execution_env.contains_key(*key))
+            .map(|s| s.to_string())
+            .collect()
+    }
+
+    pub fn output_header(is_strict: bool, color_config: ColorConfig) {
+        let ci = Vendor::get_constant().unwrap_or("unknown");
+
+        let strict_message = if is_strict {
+            "These variables WILL NOT be available to your application and may cause your build to \
+             fail."
+        } else {
+            "These variables WILL NOT be considered in your cache key and could cause inadvertent \
+             cache hits."
+        };
+
+        match ci {
+            "VERCEL" => {
+                cprintln!(
+                    color_config,
+                    BOLD,
+                    "The following environment variables are set on your Vercel project, but \
+                     missing from \"turbo.json\". {}",
+                    strict_message
+                );
+            }
+            _ => {
+                cprintln!(
+                    color_config,
+                    BOLD,
+                    "The following environment variables are missing from \"turbo.json\". {}",
+                    strict_message
+                );
+            }
+        }
+
+        let docs = color!(
+            color_config,
+            UNDERLINE,
+            "https://turbo.build/repo/docs/platform-environment-variables"
+        );
+        cprintln!(color_config, GREY, "Learn more at {docs}\n");
+    }
+
+    pub fn output_for_task(
+        missing: Vec<String>,
+        task_id_for_display: &str,
+        color_config: ColorConfig,
+    ) {
+        cprintln!(color_config, YELLOW, "{}", task_id_for_display);
+        for key in missing {
+            cprint!(color_config, GREY, "  - ");
+            cprint!(color_config, GREY, "{}\n", key);
+        }
+        println!();
```

Comment:
These are all printing to `stdout` instead of `stderr`

Reply:
Fixed!